### ISSUE TITLE
[Bug] Broken vector topology crashes wxGUI

### ISSUE THREAD

**Issue by neteler** [OSGeo/grass#488]
**Describe the bug**

Opening a vector map (polygons) with broken topology (due to previous crash), the entire wxGUI shuts down:

```
Coor file of vector map
<high_low_suitable_Europe@figures> is larger
than it should be (180 bytes excess)
Traceback (most recent call last):
  File "/home/mneteler/software/grass78_git/dist.x86_64-pc-
linux-gnu/gui/wxpython/mapwin/buffered.py", line 593, in
OnPaint

self.pdcVector.DrawToDCClipped(gcdc, rgn)
wx._core
.
wxAssertionError
:
C++ assertion "n > 1" failed at
../src/common/graphcmn.cpp(742) in StrokeLines():
Traceback (most recent call last):
  File "/home/mneteler/software/grass78_git/dist.x86_64-pc-
linux-gnu/gui/wxpython/mapwin/buffered.py", line 593, in
OnPaint

self.pdcVector.DrawToDCClipped(gcdc, rgn)
wx._core
.
wxAssertionError
:
C++ assertion "n > 1" failed at
../src/common/graphcmn.cpp(742) in StrokeLines():
Traceback (most recent call last):
[...]
```

It seems to be in a loop, repeating the same Traceback multiple times (may also be due to zooming attempts).

**To Reproduce**

Hard to reproduce. It usually silently crashes.

**System description (please complete the following information):**

```
Linux Fedora 31

g.version -rge
version=7.8.3dev
date=2020
revision=ecc6e6f79
build_date=2020-04-04
build_platform=x86_64-pc-linux-gnu
build_off_t_size=8
libgis_revision=340c2a4a0
libgis_date="Thu Apr 2 13:03:21 2020 +0200"
proj=5.2.0
gdal=2.3.2
geos=3.7.1
sqlite=3.30.0
```

**Screenshot**

![image](https://user-images.githubusercontent.com/1295172/78462872-b6318900-76d6-11ea-90cc-86e0cdc09c45.png)

**Comment by neteler** [comment]
Confirmed also for master.

**Comment by petrasovaa** [comment]
I can't reproduce it, which is not a big surprise. Maybe it happens with a particular vector you could share?

**Comment by neteler** [comment]
I have stored the vector map here as a "pack" file (EPSG 4326):

https://data.neteler.org/tmp/high_low_suitable_Europe.pack

**Comment by tmszi** [comment]
I confirm this bug (tested with `santana3a_high_low_suitable_Europe` vector map layer)

![vdigit](https://user-images.githubusercontent.com/50632337/82994636-c6912080-a002-11ea-9752-3d6bdefcf1c9.jpeg)

Error message (star editting):

```
Traceback (most recent call last):
  File "/usr/lib64/grass79/gui/wxpython/mapwin/buffered.py",
line 593, in OnPaint

self.pdcVector.DrawToDCClipped(gcdc, rgn)
wx._core
.
wxAssertionError
:
C++ assertion "n > 1" failed at /var/tmp/portage/x11-libs/wx
GTK-3.0.4-r302/work/wxWidgets-3.0.4/src/common/graphcmn.cpp(
742) in StrokeLines():
```

**Comment by neteler** [comment]
Still the case in current master:

```
GRASS version: 7.9.dev                                                          
Code revision: 64527a73f                                                        
Build date: 2020-08-06                                                          
Build platform: x86_64-pc-linux-gnu                                             
GDAL: 3.0.4                                                                     
PROJ: 6.3.2                                                                     
GEOS: 3.8.1                                                                     
SQLite: 3.32.3                                                                  
Python: 3.8.5                                                                   
wxPython: 4.0.7                                                                 
Platform: Linux-5.6.16-300.fc32.x86_64-x86_64-with-glibc2.2.5
```

**Comment by tmszi** [comment]
Additional info:

If I zoom into `santana3a_high_low_suitable_Europe` vector map layer, and start editing after that, editing work correctly. If I zoom out during editing I got same error message as mentioned above and render maps stop.

**Comment by neteler** [comment]
Great, PR #911 solves it nicely:

![image](https://user-images.githubusercontent.com/1295172/143251465-157fa758-2488-44ea-8ae8-39a0848dab17.png)

... "Yes" leads to:

![image](https://user-images.githubusercontent.com/1295172/143251581-b22ceee4-3d86-409c-b335-180242cf1aaf.png)

Thanks, @tmszi !

**Comment by petrasovaa** [comment]
Fixed in #1989, it is in main and 8.0 branch now.
